internal/api/schemas/v1.yaml

```diff
@@ -35,13 +186,99 @@ components:
       in: header
       name: X-Consul-Token
   schemas:
-    GatewayList:
+    GatewayPage:
       type: object
-      properties: {}
+      required:
+      - gateways
+      properties:
+        gateways:
+          type: array
+          items:
+            "$ref": "#/components/schemas/Gateway"
+          minItems: 0
+        meta:
+          type: object
+    Gateway:
+      type: object
+      required:
+      - name
+      - namespace
+      - listeners
+      properties:
+        name:
+          type: string
+          minLength: 1
+        namespace:
+          type: string
+          minLength: 1
+        listeners:
+          type: array
+          items:
+            "$ref": "#/components/schemas/Listener"
+          minItems: 1
+        meta:
+          type: object
+    Listener:
+      type: object
+      required:
+      - hostname
```

Comment:
Do we want to require `hostname` be defined in order to create? IIRC, this is not required in K8s land

Reply:
I believe it _is_ required in K8s land, it just defaults to `*`, which I figure we should do as well.